python/TestHarness/TestHarness.py

```diff
@@ -657,53 +450,52 @@ def handleTestResult(self, tester, output, result, start=0, end=0, add_to_table=
         # result is normally populated by a tester object when a test has failed. But in this case
         # checkRunnableBase determined the test a failure before it even ran. So we need to set the
         # results here, so they are printed if the extra_info argument was supplied
-        elif status == tester.bucket_deleted:
+        elif tester.isDeleted():
             result = tester.getStatusMessage()
 
         # Only add to the test_table if told to. We now have enough cases where we wish to print to the screen, but not
         # in the 'Final Test Results' area.
-        if add_to_table:
+        if add_to_table and self.options.checkStatus is not False: # checkStatus could be None
             self.test_table.append( (tester, output, result, timing, start, end) )
-            if status == tester.bucket_skip:
+            if tester.isSkipped():
                 self.num_skipped += 1
-            elif status == tester.bucket_success:
+            elif tester.didPass():
                 self.num_passed += 1
-            elif status == tester.bucket_pending or status == tester.bucket_pbs:
+            elif tester.isPending():
                 self.num_pending += 1
             else:
-                # Dump everything else into the failure status (neccessary due to PBS launch failures
-                # not being stored in the tester status bucket)
+                # Dump everything else into the failure status
                 self.num_failed += 1
 
         self.postRun(tester.specs, timing)
 
         print printResult(tester, result, timing, start, end, self.options)
 
-        if self.options.verbose or (not did_pass and not self.options.quiet):
+        if self.options.verbose or (self.options.quiet is not True and (tester.didFail() or tester.didDiff())):
```

Comment:
Because a test can now finish with a pending status. We don't want to treat those as errors...